### DISCUSSION TITLE
Move DHL Express over to REST calls?

### DISCUSSION THREAD

**Discussion by wooddani** [alexeybusygin/ShippingRates#35] [General]
According to the documentation at https://xmlportal.dhl.com/faq they are moving to a new approach. What are the plans on doing this? I am going to implement a solution regardless, but love the frame work here and wonder about any current work in this direction.

**Comment by alexeybusygin** [comment]
We have not started anything in this direction. But will probably start later this year. It would be great if you want to contribute!

**Comment by ebaj** [comment]
We would also like to use the REST interface. However, I see that this requirement was already submitted in May 2022. Is there still any prospect of extending the DHL interface to include REST?

**Comment by alexeybusygin** [comment]
To work on the DHL REST API implementation, I would need API credentials. Unfortunately, unlike USPS or FedEx, DHL does not provide API credentials unless you have a DHL Express customer account. Since I don't have one, I can't start the development.

If you'd like to provide credentials for development, please let me know.

**Reply by ebaj** [reply]
I only have the business login details and no test accounts. Access is now working after I completed the DHL registration, including assigning a SiteId as requested.

**Comment by ebaj** [comment]
After taking a closer look at the DHL classes, I saw that the DHL provider uses the units IN, LB, and USD. The metric system and the PreferredCurrencyCode EUR should be supported here. Could You please check and fix the units of measure currency? Best regards
